src/note/mock/note.repository.mock.ts

```diff
@@ -0,0 +1,16 @@
+import { Note } from '../note.entity';
+
+export class MockNoteRepository {
+  public async findOneOrFail({ id }: { id: number }): Promise<Note> {
+    if (id === 1) {
+      return Promise.resolve({
+        id: 1,
+        name: 'Test Note',
+        content: 'Test Note Content',
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      } as unknown as Note);
```

Comment:
Wieso wird hier eigtl das `as unknown` benötigt? Funktioniert einfach nur `{ ... } as Note` nicht?

Reply:
In dem Fall tatsächlich nicht, hast recht

Reply:
Kannst du das in deinen anderen PRs auch noch kurz überprüfen, ob das `unknown` nicht überflüssig ist, sieht bissl komisch aus

Reply:
Ja teilweise war's notwendig weil z.B. die Categories von nem Nutzer eig. ja ne Collection sind ich die in den Tests aber als Array habe. Habe dann beim Rest nicht mehr drüber nachgedacht und es immer so gemacht aber ich schaue nochmal wo man's brauch und wo nicht.